tools/parse_baselines/providers/iperf3.py

```diff
@@ -0,0 +1,106 @@
+        self._data = nesteddict()
+
+    # pylint: disable=R0201
+    def _calculate_baseline(self, data: List[float]) -> dict:
+        """Return the target and delta values, given a list of data points."""
+        avg = statistics.mean(data)
+        stddev = statistics.stdev(data)
+        return {
+            'target': math.ceil(round(avg, 2)),
+            # We add a 3% extra margin, to account for small variations that
+            # were not caught while gathering baselines. This provides
+            # slightly better reliability, while not affecting regression
+            # detection.
+            'delta_percentage': math.ceil(round(3 * stddev/avg * 100, 2)) + 3
+        }
+
+    def _format_baselines(self) -> List[dict]:
+        """Return the computed baselines into the right serializable format."""
+        baselines = dict()
+
+        for cpu_model in self._data:
+            baselines[cpu_model] = {
+                'model': cpu_model, **self._data[cpu_model]}
+
+        temp_baselines = baselines
+        baselines = []
+
+        for cpu_model in self._data:
+            baselines.append(temp_baselines[cpu_model])
+
+        return baselines
+
+    def parse(self) -> dict:
+        """Parse the rows based on proposed baselines."""
+        line = next(self._data_provider)
+        while line:
+            json_line = json.loads(line)
+            measurements = json_line['results']
+            cpu_model = json_line['custom']['cpu_model']
+
+            # Consume the data and aggregate into lists.
+            for tag in measurements.keys():
+                for key in self._baselines_defs:
+                    [ms_name, st_name] = key.split("/")
+                    ms_data = measurements[tag].get(ms_name)
+
+                    st_data = ms_data.get(st_name)
+
+                    [kernel_version, rootfs_type,
+                        iperf_config] = tag.split("/")
+
+                    data = self._data[cpu_model][ms_name]
+                    data = data[kernel_version][rootfs_type]
+                    if isinstance(data[iperf_config], list):
+                        data[iperf_config].append(st_data)
+                    else:
+                        data[iperf_config] = [st_data]
+            line = next(self._data_provider)
+
+        # Compute the baselines.
+        for cpu_model in self._data:
+            for baseline in self._data[cpu_model]:
+                data = self._data[cpu_model]
+                for kernel in data[baseline]:
+                    data = self._data[cpu_model][baseline]
+                    for rootfs in data[kernel]:
+                        data = self._data[cpu_model][baseline][kernel]
+                        for iperf_config in \
+                                data[rootfs]:
+                            data[rootfs][iperf_config] = \
+                                self._calculate_baseline(
+                                    data[rootfs][iperf_config])
```

Comment:
`data` is being reused and it's kind of confusing.

Reply:
we can indeed avoid specifying data, but since this is a very deep nested dict, we'll get pylint angry at the line length.
I can't think of a cleaner solution here

Reply:
Actually, I can solve this problem by recursion, which should be cleaner and future-proof to adding another dimension to the dict

Reply:
done